custom/shared/contexts/UIStateProvider.js

```diff
@@ -21,13 +21,29 @@ export const UIStateProvider = ({
   children,
 }) => {
   const [pinnedId, setPinnedId] = useState(null);
+  const [isMobile, setIsMobile] = useState(false);
   const [preferredViewMode, setPreferredViewMode] = useState(VIEW_MODE_SPEAKER);
   const [viewMode, setViewMode] = useState(preferredViewMode);
   const [isShowingScreenshare, setIsShowingScreenshare] = useState(false);
   const [showParticipantsBar, setShowParticipantsBar] = useState(true);
   const [showAside, setShowAside] = useState();
   const [activeModals, setActiveModals] = useState({});
   const [customCapsule, setCustomCapsule] = useState();
+  const [showAutoplayFailedModal, setShowAutoplayFailedModal] = useState(false);
+
+
+  /**
+   * Decide on view mode based on input conditions.
+   */
+  useEffect(() => {
+    if (isMobile) {
+      setViewMode(VIEW_MODE_MOBILE);
+    } else if (pinnedId || isShowingScreenshare) {
```

Comment:
Why do we have a mobile mode if we say in the README we don't support mobile layouts yet?

Reply:
oh, actually I was updating the contexts and the other stuff to be in sync with prebuilt and I just added this as in the future we will be working on making the demos mobile responsive.